src/CommunityToolkit.Maui/Behaviors/PlatformBehaviors/Touch/TouchBehavior.macios.cs

```diff
@@ -0,0 +1,327 @@
+
+		if (((platformView as IVisualNativeElementRenderer)?.Control ?? platformView) is UIButton button)
+		{
+			button.AllTouchEvents += PreventButtonHighlight;
+			((TouchUITapGestureRecognizer)touchGesture).IsButton = true;
+		}
+
+		platformView.AddGestureRecognizer(touchGesture);
+
+		if (UIDevice.CurrentDevice.CheckSystemVersion(13, 0))
+		{
+			hoverGesture = new UIHoverGestureRecognizer(async () => await OnHover(CancellationToken.None));
+			platformView.AddGestureRecognizer(hoverGesture);
+		}
+
+		platformView.UserInteractionEnabled = true;
+	}
+
+	/// <summary>
+	/// Detaches the behavior from the platform view.
+	/// </summary>
+	/// <param name="bindable">Maui Visual Element</param>
+	/// <param name="platformView">Native View</param>
+	protected override void OnDetachedFrom(VisualElement bindable, UIView platformView)
+	{
+		if (((platformView as IVisualNativeElementRenderer)?.Control ?? platformView) is UIButton button)
+		{
+			button.AllTouchEvents -= PreventButtonHighlight;
+		}
+
+		if (touchGesture is not null)
+		{
+			platformView.RemoveGestureRecognizer(touchGesture);
+			touchGesture?.Dispose();
+			touchGesture = null;
+		}
+
+		if (hoverGesture is not null)
+		{
+			platformView.RemoveGestureRecognizer(hoverGesture);
+			hoverGesture?.Dispose();
+			hoverGesture = null;
+		}
+
+		Element = null;
+	}
+
+	static void PreventButtonHighlight(object? sender, EventArgs args)
+	{
+		if (sender is not UIButton button)
+		{
+			throw new ArgumentException($"{nameof(sender)} must be Type {nameof(UIButton)}", nameof(sender));
+		}
+
+		button.Highlighted = false;
+	}
+
+	async ValueTask OnHover(CancellationToken token)
+	{
+		if (IsDisabled)
+		{
+			return;
+		}
+
+		switch (hoverGesture?.State)
+		{
+			case UIGestureRecognizerState.Began:
+			case UIGestureRecognizerState.Changed:
+				await HandleHover(HoverStatus.Entered, token);
+				break;
+			case UIGestureRecognizerState.Ended:
+				await HandleHover(HoverStatus.Exited, token);
+				break;
+		}
+	}
+
+	partial void PlatformDispose()
+	{
+		touchGesture?.Dispose();
+		touchGesture = null;
+		
+		hoverGesture?.Dispose();
+		hoverGesture = null;
+	}
+
+	sealed class TouchUITapGestureRecognizer : UIGestureRecognizer
+	{
+		readonly TouchBehavior behavior;
+
+		float? defaultRadius;
+		float? defaultShadowRadius;
+		float? defaultShadowOpacity;
+		CGPoint? startPoint;
+
+		bool isCanceled;
+
+		public TouchUITapGestureRecognizer(TouchBehavior behavior)
+		{
+			this.behavior = behavior;
+			CancelsTouchesInView = false;
+			Delegate = new TouchUITapGestureRecognizerDelegate();
+		}
+
+		public bool IsButton { get; set; }
+
+		public override async void TouchesBegan(NSSet touches, UIEvent evt)
+		{
+			base.TouchesBegan(touches, evt);
+
+			if (behavior.IsDisabled)
+			{
+				return;
+			}
+
+			isCanceled = false;
+			startPoint = GetTouchPoint(touches);
+
+			await HandleTouch(TouchStatus.Started, CancellationToken.None, TouchInteractionStatus.Started);
```

Comment:
Since this is the last line of the method, we don't care where the continuation will run.
```suggestion
			await HandleTouch(TouchStatus.Started, CancellationToken.None, TouchInteractionStatus.Started).ConfigureAwait(false);
```

Reply:
You're absolutely correct, but let's avoid this since a future PR could add more code below this line at the end of the method and cause bigger headaches than we earn from this optimization.